web/src/lib/components/asset-viewer/editor/editor-panel.svelte

```diff
@@ -0,0 +1,87 @@
+<script lang="ts">
+  import { websocketEvents } from '$lib/stores/websocket';
+  import { type AssetResponseDto } from '@immich/sdk';
+  import { mdiClose, mdiCropRotate } from '@mdi/js';
+  import { createEventDispatcher, onMount } from 'svelte';
+  import CircleIconButton from '../../elements/buttons/circle-icon-button.svelte';
+  import { t } from 'svelte-i18n';
+  import CropComponent from './crop-tool/crop-tool.svelte';
+  import Button from '$lib/components/elements/buttons/button.svelte';
+  import { cropSettingsChanged } from '$lib/stores/asset-editor.store';
+  import { derived } from 'svelte/store';
+
+  export let asset: AssetResponseDto;
+
+  onMount(() => {
+    return websocketEvents.on('on_asset_update', (assetUpdate) => {
+      if (assetUpdate.id === asset.id) {
+        asset = assetUpdate;
+      }
+    });
+  });
+
+  const dispatch = createEventDispatcher<{
```

Comment:
We are moving away from `createEventDispatcher` in favor of exposing callback function as props. You can see an example here https://github.com/immich-app/immich/blob/21d3f248dab47a3a7f835e8eb2c576a45aa37de9/web/src/lib/components/shared-components/context-menu/menu-option.svelte#L11

Reply:
done